pyproject.toml

```diff
@@ -5,6 +5,9 @@ extend-exclude = "^/src/nitsm/_pinmapinterfaces.py"
 
 [tool.ni-python-styleguide]
 extend_exclude = "src/nitsm/_pinmapinterfaces.py"
+# ignore codes not relevant to the Google docstring conventions (http://www.pydocstyle.org/en/stable/error_codes.html#default-conventions)
```

Comment:
Why is this project using different conventions than the ni style guide (if I'm understanding this correctly)?

Reply:
The ni style guide doesn't specify a docstring convention to use, but the Google style is recommended.  Since the style guide doesn't enforce a docstring convention they run `pydocstyle` against all checks:
https://github.com/ni/python-styleguide/blob/d91d60257add221fbf1e10267b00be459946c84b/ni_python_styleguide/config.ini#L88
There are certain checks that only pertain to certain styles as outlined here:
http://www.pydocstyle.org/en/stable/error_codes.html#default-conventions
Since we decided on the Google style for this project this PR ignores all the checks that don't apply.  Furthermore, the Google style requires a one line description followed by a blank line then a multiline summary (corresponds to error code D415). Since we copy most of our docs from the .NET documentation this rule is often broken.  Thus, I thought it would be prudent to ignore it so the python docstrings don't stray too far from the .NET docs.

Reply:
Gotcha. That makes sense to me.